reactive/docker.py

```diff
@@ -74,67 +89,89 @@ def install():
     ]
     apt_update()
     apt_install(packages)
+
     # Install docker-engine from apt.
-    if config('install_from_upstream'):
+    if config('docker_runtime') == "upstream" or config('install_from_upstream'):
         install_from_upstream_apt()
+    elif config('docker_runtime') == "nvidia":
+        install_from_nvidia_apt()
     else:
         install_from_archive_apt()
 
+    hookenv.log('pre docker_runtime={0}'.format(config('docker_runtime')))
     validate_config()
     opts = DockerOpts()
-    render('docker.defaults', '/etc/default/docker', {'opts': opts.to_s()})
+    hookenv.log('post docker_runtime={0}'.format(config('docker_runtime')))
+    render('docker.defaults', '/etc/default/docker', {'opts': opts.to_s(),'docker_runtime': config('docker_runtime') })
     render('docker.systemd', '/lib/systemd/system/docker.service', config())
     reload_system_daemons()
 
-    apt_hold(['docker-engine'])
-    apt_hold(['docker.io'])
+    holdall()
     hookenv.log('Holding docker-engine and docker.io packages' +
                 ' at current revision.')
 
+    host.service_restart('docker')
     hookenv.log('Docker installed, setting "docker.ready" state.')
     set_state('docker.ready')
 
     # Make with the adding of the users to the groups
     check_call(['usermod', '-aG', 'docker', 'ubuntu'])
 
 
-@when('config.changed.install_from_upstream', 'docker.ready')
+#@when('config.changed.install_from_upstream', 'docker.ready')
+@when('config.changed.docker_runtime', 'docker.ready')
 def toggle_docker_daemon_source():
-    ''' A disruptive toggleable action which will swap out the installed docker
+    ''' A disruptive 
```

Comment:
Maybe clean this comment up a little? A disruptive reaction to config changing that will ...